app/com/linkedin/drelephant/tuning/engine/SparkExecutionEngine.java

```diff
@@ -74,6 +74,19 @@ public void parameterOptimizerIPSO(List<AppResult> results, JobExecution jobExec
 
   @Override
   public String parameterGenerationsHBT(List<AppResult> results, List<TuningParameter> tuningParameters) {
+    if (results != null && results.size() > 0) {
+      SparkHBTParamRecommender sparkHBTParamRecommender = new SparkHBTParamRecommender(results.get(0));
```

Comment:
Suggestion: Leave a comment indicating that we take only the first entry in AppResult because a job execution for Spark would correspond to only one app.